Adrian AI Materials (#5)
Ada-Developers-Academy/independent-ai-topics: working-with-openai/setting-up-openai.md

> ## Setting up the OpenAI API

Reviewer: Removed a hashtag to make the title an H1.

```suggestion
# Setting up the OpenAI API
```

Reviewer: I don't feel that the title has to change, but I had the thought that something like "Accessing the OpenAI APIs" or "Setting up OpenAI API Access" would be just a little closer/clearer about what is covered.

## Goals

> Now that we have a good idea of what the OpenAI API is and some of the ways we can use it, let's go ahead and get it set up. We'll set up our accounts and then set everything up for us to start integrating the OpenAI API into our projects!

Reviewer: Some potential word smithing to reduce how many times we use the phrase "set up" in this paragraph.

```suggestion
Now that we have a good idea of what the OpenAI API is and some of the ways we can use it, let's go ahead and get it set up. We'll create or configure our account and then confirm that everything is in place for us to start integrating the OpenAI API into our projects!
```

   *Fig. OpenAI Phone verification code input*  

### !callout-warning

## Credits Warning

If your phone number has already been used to verify an OpenAI account, you will receive the following warning:  

![OpenAI credits warning](assets/setting-up-openai/open-ai-credits-warning.png)  
*Fig. OpenAI credits warning*

### !end-callout

4. Create your secret key. 
   - Now that you have verified your phone number, you can create a secret key. On the API Keys screen, press the button that says "+ Create new secret key" and a pop up will appear to prompt you to create a new secret key.  
   ![Secret key pop up](assets/setting-up-openai/open-ai-secret-key-pop-up.png)  
   *Fig. OpenAI secret key creation pop up*

   - Once you press "Create secret key", you will be taken to another pop up with a prompt to save your key in a secure location. This will be your only chance to copy your key, so make sure to copy it over somewhere safe so that you can use it later.  
   ![Save secret key pop up](assets/setting-up-openai/open-ai-save-secret-key.png)  
   *Fig. Pop up window with alert to save secret key*

> 5. Verifying credits were added

Reviewer: Since "Creating Your OpenAI Account" & "Creating an API Key" are sections that start with an H2 heading, it might be more consistent and help break up the page to treat this "Verifying credits were added" section similarly.

Author: Done!

   - If you did not receive a warning about credits, you can verify your credits have been added by navigating to the "Usage" tab in the menu on the left side of the page.  
   ![Usage tab in OpenAI menu](assets/setting-up-openai/open-ai-usage-tab.png)  
   *Fig. OpenAI side menu with "usage" tab circled*  

   - You will be taken to a page and the panel on the right side will have a list of the credits you have recieved and the credits you have used. Here you can verify that credits have been added to your account.     
   ![OpenAI Usage statistics example](assets/setting-up-openai/open-ai-credits-verification.png)
   *Fig. OpenAI usage page with lists of gredits granted and credits used.*  
   - If you do not see credits associated with your account, feel free to reach out to an instructor for assistance.

### !callout-info

## Credits Expiration

> In the image above, you may notice that some of the credits have expired. The free credits granted from OpenAI expire after 3 months. You should have received $5.00 which will be more than enough to complete the coursework for the next few lessons with plenty left over to play around on your own. Information on the OpenAI pricing structure can be found [here](https://openai.com/pricing).

Reviewer: Plenty is relative since it really depends on what students are thinking of trying out, so I'm thinking we could be more general here?

```suggestion
In the image above, you may notice that some of the credits have expired. The free credits granted from OpenAI expire after 3 months. You should have received $5.00 which will be more than enough to complete the coursework for the next few lessons with some left over to play around on your own. Information on the OpenAI pricing structure can be found [here](https://openai.com/pricing).
```
